src/Spectre.libWavelet.Tests/WaveletDecomposerRefTest.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+* WaveletDecomposerRefTest.cpp
+* Tests whether the signal is properly decomposed by the reference 
+* implementation.
+*
+Copyright 2018 Michal Gallus
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <gtest/gtest.h>
+#include "Spectre.libWavelet\WaveletDecomposerRef.h"
+
+namespace
+{
+using namespace spectre::algorithm::wavelet;
+
+static void CompareVectors(const std::vector<DataType>& estimate,
```

Comment:
Isn't that already in GTest? Check [this](https://stackoverflow.com/a/2797990/3067956) and [this](https://github.com/google/googletest/blob/master/googlemock/docs/CheatSheet.md#floating-point-matchers). It should also provide better reporting of failure.

Reply:
I do not really see a way to achieve that with the default functionality.
In this [answer](https://stackoverflow.com/a/13112240) the author actually implements his own matcher for floating point array comparison

Reply:
As we spoke, I suppose it will be implemented with my matcher chunk into this review?

Reply:
It will